server/src/test/java/io/crate/integrationtests/PartitionedTableConcurrentIntegrationTest.java

```diff
@@ -441,4 +443,51 @@ public void testConcurrentPartitionCreationWithColumnCreationTypeMissMatch() thr
         // this query should never fail
         execute("select * from t1 order by x limit 50");
     }
+
+    @Test
+    public void test_stress_partition_deletion_and_creation() throws Exception {
+        execute(
+            "create table tbl (x int, p int) " +
+            "clustered into 2 shards partitioned by (p) with (number_of_replicas = 0)");
+        int numOpsPerThread = randomIntBetween(100, 300);
+        int numInsertThreads = randomIntBetween(5, 10);
+        int numDeleteThreads = randomIntBetween(5, 10);
+        Phaser phaser = new Phaser(numInsertThreads + numDeleteThreads);
+        List<Thread> threads = new ArrayList<>(numInsertThreads + numDeleteThreads);
+        for (int t = 0; t < numInsertThreads; t++) {
+            Thread thread = new Thread(() -> {
+                phaser.arriveAndAwaitAdvance();
+                for (int index = 0; index < numOpsPerThread; index++) {
+                    int numRows = randomIntBetween(1, 10);
+                    Object[][] rows = new Object[numRows][];
+                    for (int i = 0; i < numRows; i++) {
+                        rows[i] = new Object[] { index, randomIntBetween(1, 10) };
+                    }
+                    execute("insert into tbl (x, p) values (?, ?)", rows);
+                }
+            });
+            thread.start();
+            threads.add(thread);
+        }
+        for (int t = 0; t < numDeleteThreads; t++) {
+            Thread thread = new Thread(() -> {
+                phaser.arriveAndAwaitAdvance();
+                for (int index = 0; index < numOpsPerThread; index++) {
+                    int partition = randomIntBetween(1, 10);
```

Comment:
Is this "safe", it can also include partitions that are not created.

Reply:
that's part of the idea of the test :)